pom.xml

```diff
@@ -620,50 +684,19 @@
                 <goal>jar</goal>
               </goals>
             </execution>
-            <execution>
-              <id>default-test-jar</id>
```

Comment:
IMO such a dependency is a bug, for sure. Factoring out into a test utility is best, if possible. But here is a case where we cannot:

 - class A depends on SDK main
 - SDK test depends on class A
 - non-SDK test depends on class A

In this case, class A must be in SDK main. I think many things fall into this category.